### PR TITLE
Refactor validateSubmit in admin_predicate_configuration.ts

### PR DESCRIPTION
**Pull request by jeffpw-goog**
### Description

* Remove unnecessary `Array.from`s
* Change ? to ! because #predicate-config-value-row-container should always exist and is an error if it doesn't, rather than failing silently
* Condition the missing value logic on whether the input is a checkbox type, rather than whether there is more than one input
* Remove unnecessary early returns that are barely optimizing anything and add complexity
* Re-work the logic to show the error message to flow more naturally

In addition to being cleaner this enables #6109 where we will have more than one non-checkbox input box.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [X] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [X] Performed manual testing (Chrome and Firefox if it includes front-end changes)

### CODE REVIEW

File: server/app/assets/javascripts/admin_predicate_configuration.ts [hunk 1]
```diff
@@ -72,72 +72,62 @@ class AdminPredicateConfiguration {
...
+      .querySelectorAll<HTMLSelectElement>('.cf-operator-select select')
+      .forEach((el: HTMLSelectElement) => {
+        if (el.options[el.options.selectedIndex].value == '') {
+          hasSelectionMissing = true
+        }
+      })
 
     // Check if any inputs are missing a value.
     document
-      .querySelector('#predicate-config-value-row-container')
```
Comment (jcma-google) [inline server/app/assets/javascripts/admin_predicate_configuration.ts:96]: nit: #predicate-config-value-row-container name isn't very clear, I'm not sure if this is a convention or not but if it isn't is it possible to rename to something clearer? Feel free to ignore if my request is naive and the cost outweighs the benefits.

**Review by jcma-google — COMMENTED** [review]
Are tests already present for this use case? Should we add any?

File: server/app/assets/javascripts/admin_predicate_configuration.ts (hunk 1 above)
Comment (jeffpw-goog) [reply server/app/assets/javascripts/admin_predicate_configuration.ts:96]: Thanks for the suggestion! Based on the naming throughout the page this name actually does make decent sense to me - this page is for predicate config, and value rows are the rows of value input boxes (vs the scalar, operator, etc. inputs at the top), and this contains all of those. The term value row is used widely throughout the page and associated utils (like for testing). I'm happy to consider if you have a specific suggestion, but I don't have a better one myself.

**Comment by jeffpw-goog** [comment]
> Are tests already present for this use case? Should we add any?

Yes there are tests in admin_predicates.test.ts that cover each of the error cases: https://github.com/civiform/civiform/blob/92045e2e92c047b3126f91c8ce1dcb4cda859e53/browser-test/src/admin/admin_predicates.test.ts#L242-L284

**Review by jcma-google — APPROVED** [review]

**Pull request merged**